storage/src/test/java/tech/pegasys/teku/storage/server/network/DatabaseNetworkTest.java

```diff
@@ -82,4 +86,45 @@ public void shouldNotThrowIfForkAndContractMatch(@TempDir final File tempDir) th
 
     assertDoesNotThrow(() -> DatabaseNetwork.init(networkFile, fork, eth1Address));
   }
+
+  @Test
+  void shouldWriteAndReadDatabaseNetworkWithChainId(@TempDir final File tempDir)
+      throws IOException {
+    final File networkFile = new File(tempDir, "network.yml");
+
+    final Bytes4 fork = dataStructureUtil.randomFork().getCurrentVersion();
+    final Eth1Address eth1Address = dataStructureUtil.randomEth1Address();
+    final Long depositId = dataStructureUtil.randomLong();
+    final DatabaseNetwork databaseNetwork =
+        new DatabaseNetwork(fork.toHexString(), eth1Address.toHexString(), depositId);
+
+    final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    objectMapper.writerFor(DatabaseNetwork.class).writeValue(networkFile, databaseNetwork);
+    final DatabaseNetwork readDatabaseNetwork =
+        objectMapper.readerFor(DatabaseNetwork.class).readValue(networkFile);
+
+    assertEquals(fork.toHexString(), readDatabaseNetwork.forkVersion);
+    assertEquals(eth1Address.toHexString(), readDatabaseNetwork.depositContract);
+    assertEquals(depositId, readDatabaseNetwork.depositChainId);
+  }
+
+  @Test
+  void shouldWriteAndReadDatabaseNetworkWithoutChainId(@TempDir final File tempDir)
+      throws IOException {
+    final File networkFile = new File(tempDir, "network.yml");
+
+    final Bytes4 fork = dataStructureUtil.randomFork().getCurrentVersion();
+    final Eth1Address eth1Address = dataStructureUtil.randomEth1Address();
+    final DatabaseNetwork databaseNetwork =
+        new DatabaseNetwork(fork.toHexString(), eth1Address.toHexString(), null);
+
+    final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    objectMapper.writerFor(DatabaseNetwork.class).writeValue(networkFile, databaseNetwork);
```

Comment:
because we're writing here with the mapper, we'll want to read the file by hand and check deposit_chain_id is not present in the file... otherwise we're not really testing what we think we are here

Reply:
just this one looks unresolved @gconnect  can we read the file nad make sure its not got that field?

Reply:
Done